zone/botspellsai.cpp

```diff
@@ -219,7 +216,7 @@ bool Bot::AICastSpell(Mob* tar, uint8 chance, uint16 spell_type, uint16 sub_targ
 
 	for (const auto& s : bot_spell_list) {
 
-		if (!IsValidSpellAndLoS(s.SpellId, HasLoS())) {
+		if (!IsValidSpellAndLoS(s.SpellId, (!BotRequiresLoSToCast(spell_type, s.SpellId) ? true : HasLoS()))) {
```

Comment:
This is extremely gnarly to read

`IsValidSpellAndLoS` should just validate a valid spell, not also check LOS

There should be a separate check that handles LOS requirements of a spell which can do ternary stuff inside it and make things a little easier to read. This is duplicated in a few other spots.